File: backend/app/routers/content.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from ..database import get_db
from ..models import PrecautionContent, ContentCategory, Role, User
from ..schemas import ContentOut, ContentCreate
from ..security import require_roles

router = APIRouter(prefix="/api/content", tags=["content"])


@router.get("", response_model=list[ContentOut])
def list_content(category: ContentCategory | None = None,
                 body_region: str | None = None,
                 db: Session = Depends(get_db)):
    q = db.query(PrecautionContent).filter(PrecautionContent.is_published == True)  # noqa: E712
    if category:
        q = q.filter(PrecautionContent.category == category)
    if body_region:
        q = q.filter(PrecautionContent.body_region == body_region)
    return q.order_by(PrecautionContent.created_at.desc()).all()


@router.get("/{content_id}", response_model=ContentOut)
def get_content(content_id: str, db: Session = Depends(get_db)):
    item = db.query(PrecautionContent).filter(PrecautionContent.id == content_id).first()
    if not item:
        raise HTTPException(status_code=404, detail="Content not found")
    return item


@router.post("", response_model=ContentOut, status_code=201)
def create_content(payload: ContentCreate,
                   user: User = Depends(require_roles(Role.doctor, Role.admin)),
                   db: Session = Depends(get_db)):
    item = PrecautionContent(**payload.model_dump())
    db.add(item)
    db.commit()
    db.refresh(item)
    return item

File: server/routes/content.js
const { Router } = require("express");
const { v4: uuid } = require("uuid");
const db = require("../db");
const { authenticate, requireRoles } = require("../auth");

const router = Router();

router.get("/", (req, res) => {
  const { category, body_region } = req.query;
  let sql = "SELECT * FROM precaution_content WHERE is_published = 1";
  const params = [];
  if (category) {
    sql += " AND category = ?";
    params.push(category);
  }
  if (body_region) {
    sql += " AND body_region = ?";
    params.push(body_region);
  }
  sql += " ORDER BY created_at DESC";
  const items = db.prepare(sql).all(...params);
  res.json(items);
});

router.get("/:id", (req, res) => {
  const item = db.prepare("SELECT * FROM precaution_content WHERE id = ?").get(req.params.id);
  if (!item) return res.status(404).json({ detail: "Content not found" });
  res.json(item);
});

router.post("/", authenticate, requireRoles("doctor", "admin"), (req, res) => {
  const { category, title, summary, body, media_type, media_url, body_region } = req.body;
  const id = uuid();
  db.prepare(
    `INSERT INTO precaution_content (id, category, title, summary, body, media_type, media_url, body_region)
     VALUES (?, ?, ?, ?, ?, ?, ?, ?)`
  ).run(id, category, title, summary || null, body || null, media_type || "article", media_url || null, body_region || null);

  const item = db.prepare("SELECT * FROM precaution_content WHERE id = ?").get(id);
  res.status(201).json(item);
});

module.exports = router;
